pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,417 @@
+    if 'CIM_Blah' in class_store:                    # test if class exists
+        klass = class_store.get('CIM_Blah;)          # get the class
+"""
+
+from abc import abstractmethod, abstractproperty
+from six import add_metaclass, PY2
+from custom_inherit import DocInheritMeta
+
+
+def compatibleabstractproperty(func):
+    """
+    Python 2 and python 3 differ in decorator for abstract property.
+    in python 3 (gt 3.3) it is:
+        @property
+        @abstractproperty
+    in python 2
+        @abstractproperty
+    """
+
+    if PY2:  # pylint: disable=no-else-return
+        return abstractproperty(func)
+    else:
+        return property(abstractmethod(func))
+
+
+@add_metaclass(DocInheritMeta(style="google", abstract_base_class=True))
+class BaseObjectStore(object):
+    """
+    An abstract class that defines the APIs for the methods of an object store
+    for CIM objects including CIMClass, CIMInstance, and CIMQualifierDeclaration
+    objectsthat constitute a WBEM server repository.  This
+    class provides the abstract methods for creating, accessing, and deleting,
+    CIM objects of a single CIM object type in the repository.
+
+    An object store stores only a single CIM object type.
+    """
+
+    @abstractmethod
+    def __init__(self, cim_object_type):
+        """
+        Instantiate the self.cim_object_type variable.
+
+        Parameters:
+
+            cim_object_type(:term:`CIM object`):
+              The Pywbem CIM object type as defined in cim_types.py for the
+              objects in the data store. Used to verify values on create.
+        """
+        self._cim_object_type = cim_object_type
+
+    @abstractmethod
+    def exists(self, name):
+        """
+        Test if the CIM object identified by name exists in the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+        Returns:
+
+            True if the object exists; False if it does not exist
+        """
+        pass
+
+    @abstractmethod
+    def get(self, name, copy=True):
+        """
+        Return the CIM object identified by name if it exists in the object
+        store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+          copy(:class:`py:bool`):
+            If True, returns a copy of the object to insure that modifying the
+            returned object does not change the data store. The default
+            behavior is True .  If copy is False, the object in the object
+            store is returned but if it is modified by the user, the object in
+            the store may be modified also.
+        Returns:
+
+            :term:`CIM object`: Returns the CIM object identified by the
+              name parameter.
```

Comment:
The indentation of the second line in the Returns section causes a paragraph break in the Sphinx output. Remove the indentation. 

Note: In the Returns section, indentation must not be there, in the Parameters section, indentation must be there).